### ISSUE TITLE
ReactImageManager.cs (memory problem with fresco.imagepipeline)

### ISSUE THREAD

**Issue by zfanda** [Microsoft/react-native-windows#1463]
### Environment
- Target Platform: 10.0.10586
- Target Platform Version(s): Mobile (ARM)

### Actual Behavior
Repeating calls of imagePipeline.FetchDecodedBitmapImageAsync(ImageRequest.FromUri(uri)) function (fresco.imagepipeline library) consumes all available application memory and application crashes  (sometimes system restarted).

### Steps to Reproduce
Load multiple large images from local folders (e.g. CameraRoll folder):
```
<Image source={{uri: localFile.uri}}  />  
<Image source={{uri: localFile2.uri}}  />  
...
...
<Image source={{uri: localFile10.uri}}  /> 
```

### Expected Behavior
When I replaced the code with this UWP code:
```
StorageFile file = await StorageFile.GetFileFromPathAsync(source);
using (StorageItemThumbnail thumbnail = await file.GetThumbnailAsync(thumbnailMode))
{
     BitmapImage bitmapImage = new BitmapImage();
     bitmapImage.SetSource(thumbnail);
     image = bitmapImage;
}
```
then the application worked fine/better (less memory usage and application not crashes).

**Comment by phongcao** [comment]
Hi @zfanda, can you please provide me some more info:

- What's the size of the image?
- How many images that you're loading?
- What's the target mobile device?

The main different here is that FetchDecodedBitmapImageAsync returns [WriteableBitmap](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.imaging.writeablebitmap) while your replaced code uses [BitmapImage](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.imaging.bitmapimage). 
BitmapImage stores encoded data and only decodes it when rendering so it consumes less memory than WriteableBitmap but has lower performance. If you have a list of 100+ images, you can see the difference while scrolling. Furthermore, if you need to edit the image, it will need to be converted to WriteableBitmap later.

The solution I'm thinking now is that we can fetch the local images as BitmapImage and it should be working fine. It would be great if you can share me some working example.

**Comment by zfanda** [comment]
_What's the size of the image?_
1.5 - 2 MB (3264 x 1840 phone camera picture)

_How many images that you're loading?_
~10 images, but application not loaded all images (app crashed earlier) 

_What's the target mobile device?_
Microsoft Lumia 650

MemoryManager.AppMemoryUsageLimit shows for my device ~400MB max. available memory.

**Comment by zfanda** [comment]
It might be good, to modify the code, so that it uses image source parameters width/height to retrieve smaller resizing image data (thumbnail). 
[https://github.com/facebook/react-native/blob/efcdef711eba82b2905d237ee9a3d094652c37ac/Libraries/Image/ImageSourcePropType.js](url)

<Image source={{uri: uri, width: 100, height:100}} style={{width: 200, height: 200}}/>

**Comment by phongcao** [comment]
Thanks, let me check and get back to you after a few days.

**Comment by phongcao** [comment]
Hi @zfanda, I'm sorry for my late reply on this issue. Could you please help try if this helps fix the memory issue?
https://github.com/phongcao/react-native-windows/tree/encoded-local-image

**Comment by rozele** [comment]
Closed by #1563
